tvm/src/codegen/llvm/codegen_llvm.cc

```diff
@@ -1009,6 +1012,13 @@ llvm::Value* CodeGenLLVM::VisitExpr_(const Call* op) {
     return CreateIntrinsic(op);
   } else if (op->call_type == Call::Extern ||
              op->call_type == Call::PureExtern) {
+    if ((op->name).compare("TVMBackendAllocWorkspace") == 0) {
+      assert_alloc_mem.push_back(assert_free());
+      assert_alloc_mem.back().dev_type = op->args[0];
+      assert_alloc_mem.back().dev_id = op->args[1];
+      assert_alloc_mem.back().depth = loop_depth;
```

Comment:
Let's create an instance first then push it back to the array. Instead of calling `.back()` multiple times.